README.md

```diff
@@ -0,0 +1,34 @@
+# x3d2: Next-Generation High-Fidelity CFD Solver
+
+x3d2 is an open-source Computational Fluid Dynamics (CFD) solver, designed as the next-generation evolution of Xcompact3d.  It leverages GPU acceleration while maintaining the high-order finite-difference methods and spectral-like accuracy of its predecessor. x3d2 is optimised for High-Performance Computing (HPC) environments and supports both CPU and GPU parallelisation.
+
+## Key Features
+* **High-Order Finite-Difference Methods:** Compact finite-difference schemes delivering spectral-like accuracy on a monobloc Cartesian mesh, ensuring high-fidelity simulations.
+* **Dual Parallelisation Backends:**
+    * **OpenMP:** Optimised for multi-core CPU execution.
+    * **CUDA:** Designed for GPU acceleration on NVIDIA hardware.
+* **Scalable Parallel Computing:** Efficient MPI-based parallelisation for Direct and Large Eddy Simulations (DNS/LES) of turbulent flows, allowing for simulations on hundreds of thousands of cores and multiple GPUs.
+* **Incompressible & Low-Mach Variable Density Flows:** Solves the Navier-Stokes equations with fractional time-stepping and spectral Poisson solvers.
+* **Modern Software Engineering:** x3d2 is being actively developed with a focus on maintainability, extensibility, and code quality.
+
+## Performance and Scalability
+
+x3d2 is built for high-performance computing and is designed to scale efficiently on modern HPC architectures.  Its performance is enhanced by:
+
+* **DistD2-TDS:** A novel algorithm for solving tridiagonal systems with improved data locality and minimal data movement. This algorithm is specifically optimised for both CPUs (using vectorisation) and GPUs (using thread-level parallelism) and significantly reduces inter-process communication.
+* **2D Domain Decomposition for FFT:** x3d2 utilises 2D domain decomposition for Fast Fourier Transforms (FFTs), enabling greater scalability and allows for efficient parallelisation of the Poisson solver, especially for large simulations. x3d2 uses optimised FFT implementations (cuFFT for GPUs and 2DECOMP&FFT for CPUs) to accelerate the solution of the Poisson equation.
```

Comment:
cuFFT is single GPU only so to avoid confusion please replace with "cuFFTMp library on NVIDIA GPUs and 2DECOMP&FFT on CPUs."